### PR TITLE
Fix #11568

### LINKED ISSUE #11568: Truncate html text with style tag
* [x] bug
* [x] enhancement

CakePhp v 3.5.8

When i truncate html text with style tag, the content of style tag is counted.
So the returned string contain only style tag content because my desired length is shoten than style tag content.
It's normaly or it's a forgotten (so i can propose a pull request) ?

### PR DESCRIPTION
**Pull request by maratth**
- Added var for default excluded html tags and getter/setter
- Added check in truncate html tags loop, for seeing if tag is excluded or not

### CODE REVIEW

File: src/Utility/Text.php [hunk 1]
```diff
@@ -1120,4 +1133,25 @@ public static function slug($string, $options = [])
...
+    }
+
+    /**
+     * Set html tags not counted for truncate.
+     *
+     * @param array $htmlNoCount New html tags not counted for truncate.
+     * @return void
+     */
+    public static function setHtmlNoCount($htmlNoCount)
+    {
```
Comment (markstory) [inline src/Utility/Text.php]: Can we have a typehint here to ensure the parameter is an array?

**Review by markstory — COMMENTED** [review]
Do we need the new methods? Adding new methods would push this into 3.6.

File: src/Utility/Text.php [hunk 2]
```diff
@@ -608,7 +618,10 @@ public static function truncate($text, $length = 100, array $options = [])
 
             preg_match_all('/(<\/?([\w+]+)[^>]*>)?([^<>]*)/', $text, $tags, PREG_SET_ORDER);
             foreach ($tags as $tag) {
-                $contentLength = self::_strlen($tag[3], $options);
+                $contentLength = 0;
+                if (!in_array($tag[2], static::$_defaultHtmlNoCount)) {
```
Comment (dereuromark) [inline src/Utility/Text.php]: best to set third param to "true" here.

File: src/Utility/Text.php (hunk 1 above)
Comment (LifeOrYou) [reply src/Utility/Text.php]: I need this into 3.5.x, i must remove methods for this time ?

File: src/Utility/Text.php (hunk 2 above)
Comment (LifeOrYou) [reply src/Utility/Text.php]: thanks for the tip

File: src/Utility/Text.php (hunk 1 above)
Comment (markstory) [reply src/Utility/Text.php]: Yes. I don't think we need to make these tags configurable.

**Comment by markstory** [comment]
Thanks!

**Pull request merged**